spec/models/mark_spec.rb

```diff
@@ -155,6 +155,7 @@
     end
 
     it 'does not take into account deductions related to other criteria' do
+      assignment.reload
```

Comment:
I'm not sure why this is necessary?

Reply:
So this is one of the things I was unsure about. Yesterday, when I added the validation that a category cannot belong to a criterion of another assignment, this test failed. When I went through it with byebug it seems that the call to create immediately after it is troublesome, because of the execution. The `assignment` that is declared in the `let` before the examples is referenced and thus created within this other create for a flexible_criterion, and when the category created within the creation of the new flexible criterion does not seem to be aware the the criterion it belongs to is in the list of the assignment's criteria, because the criterion is still being processed.

Reply:
So this is one of the things I was unsure about. Yesterday, when I added the validation that a category cannot belong to a criterion of another assignment, this test failed. When I went through the test with byebug, the call to `create(:flexible_criterion_with_annotation_category,  assignment: assignment)` immediately after my addition stops the test due to failed validation of an annotation_category (before I added the `reload`) I believe this happens because of the order of execution within that create. In the create, the assignment that is declared in the `let` (before the all the examples, including this test) is referenced and thus created. When the annotation category created as a consequence of the creation of the new flexible criterion validates, the category's predicate that checks the list of criteria it can belong to does not find that the criteria it belongs is in the assignment's list of criteria, because the criterion is still being processed.

Reply:
So this is one of the things I was unsure about. Yesterday, when I added the validation that a category cannot belong to a criterion of another assignment, this test failed. When I went through the test with byebug, the call to `create(:flexible_criterion_with_annotation_category,  assignment: assignment)` immediately after my addition stops the test due to failed validation of an annotation_category (before I added the `reload`). I believe this happens because of the order of execution within that create. In the create, the assignment that is declared in the `let` (before the all the examples, including this test) is referenced and thus created. When the annotation category created as a consequence of the creation of the new flexible criterion validates, the category's predicate that checks the list of criteria the category can belong to does not find that the criteria the category belongs to is in the assignment's list of criteria. Due to, I believe, the criterion is still being processed in memory alongside the assignment.